ui/app/utils/has-variable-declarations.js

```diff
@@ -0,0 +1,6 @@
+export default function hasVariableDeclarationsAndReferences(spec) {
+  const VAR_DECLARATION_AND_REFERENCE_PATTERN = '\\b(?:variable|var\\.)\\b';
```

Comment:
```suggestion
  const VAR_DECLARATION_AND_REFERENCE_PATTERN = /\b(?:variable[s]?|var\.)\b/;
```

Sorry, late to the party 😅 

There's a succinct way to declare variables that looks like this:

```hcl
variables {
  image = "redis:7"
  cpu   = 500
}
```

I think this regex will capture both? 

Using `/.../` also sidesteps the need to escape slashes, making the expression easier to read, and also compiles it automatically, so it's not necessary to call `new RegExp`. Since this pattern is static you can also declare the constant outside the function so it's computed only once instead of every time the function is called.

It's also worth keeping in mind that this can generate false positives, for example, a job like this would return `true` even though it doesn't use variables:

```hcl
job "example" {
  group "cache" {
    task "redis" {
      driver = "raw_exec"

      config {
        command = "/bin/bash"
        args    = ["${NOMAD_TASK_DIR}/script.sh"]
      }

      template {
        data = <<EOF
{{ $var := sprig_dict "a" "b" }}

while true; do
  echo "{{ $var.a }}"
  sleep 3
done
EOF

        destination = "${NOMAD_TASK_DIR}/script.sh"
      }
    }
  }
}
```

It's a pretty weird example, and you have is probably good enough depending on how this function is going to be used 👍